In [1]:
import pandas as pd
import numpy as np
from scipy.stats import ttest_ind
from scipy.stats import shapiro
from scipy import stats

In [2]:
df = pd.read_excel('21.10 CASE_LOSSES.xlsx')
df

loss_id    client_id                    loss_name  loss_payout_amt  \
0         1  10000713193  Оказание медицинской помощи           100000   
1         2    100014656  Оказание медицинской помощи            50000   
2         3  10006545463  Оказание медицинской помощи            30000   
3         4  10006611194  Оказание медицинской помощи            70000   
4         5   1001091597  Оказание медицинской помощи            60000   
5         6   1001198959  Оказание медицинской помощи            50000   
6         7    100193151  Оказание медицинской помощи           100000   
7         8  10034571331  Оказание медицинской помощи            50000   
8         9  10041665730  Оказание медицинской помощи            20000   
9        10   1004171531  Оказание медицинской помощи           100000   
10       11  10049477653  Оказание медицинской помощи           300000   
11       12   1004971034  Оказание медицинской помощи            30000   
12       13  10053883633  Оказание медицинской помощи            50000   
13       14    100543063  Оказание медицинской помощи            40000   
14       15    100639851  Оказание медицинской помощи           150000   
15       16    100655800  Оказание медицинской помощи            50000   
16       17   1009008749  Оказание медицинской помощи            90000   
17       18   1010518468  Оказание медицинской помощи            50000   
18       19     10107896  Оказание медицинской помощи            20000   
19       20    101095518  Оказание медицинской помощи            50000   
20       21  10109591977  Оказание медицинской помощи           100000   
21       22   1011049180  Оказание медицинской помощи            50000   
22       23   1011175413  Оказание медицинской помощи          1000000   
23       24  10111991766  Оказание медицинской помощи          1500000   
24       25  10118841571  Оказание медицинской помощи           200000   
25       26  10139699711  Оказание медицинской помощи           250000   
26       27  10146335363  Оказание медицинской помощи           150000   
27       28  10151551961  Оказание медицинской помощи           100000   
28       29   1015458577  Оказание медицинской помощи           100000   
29       30   1015458839  Оказание медицинской помощи            50000   
30       31  10159717564  Оказание медицинской помощи           200000   
31       32     10160959  Оказание медицинской помощи            50000   
32       33  10164911618  Оказание медицинской помощи            80000   
33       34  10166187767  Оказание медицинской помощи            50000   
34       35    101664801  Оказание медицинской помощи          1000000   
35       36  10168743459  Оказание медицинской помощи           800000   
36       37  11163199740  Оказание медицинской помощи           500000   
37       38   1860164119  Оказание медицинской помощи           700000   
38       39   1000484488  Оказание медицинской помощи          1300000   
39       40  10007398134  Оказание медицинской помощи           100000   
40       41    100077669  Оказание медицинской помощи           100000   
41       42  10010634147  Оказание медицинской помощи            50000   
42       43  10011174381  Оказание медицинской помощи            80000   
43       44  10144491815  Оказание медицинской помощи            30000   
44       45   1000579095  Оказание медицинской помощи            20000   

    Unnamed: 4  Unnamed: 5  
0          NaN         NaN  
1          NaN         NaN  
2          NaN         NaN  
3          NaN         NaN  
4          NaN         NaN  
5          NaN         NaN  
6          NaN         NaN  
7          NaN         NaN  
8          NaN         NaN  
9          NaN         NaN  
10         NaN         NaN  
11         NaN         NaN  
12         NaN         NaN  
13         NaN         NaN  
14         NaN         NaN  
15         NaN         NaN  
16         NaN         NaN  
17         NaN         NaN  
18         NaN         NaN  
19         NaN     

In [3]:
df2 = pd.read_excel('21.10 CASE_CONTRACTS.xlsx')
df2

contract_id     contract_num                  product_name    client_id  \
0        21111219  ТТЕ7227715*****  Страхование путешественников  10161870404   
1        21111381  ТТЕ7227715*****  Страхование путешественников   1017159879   
2        21112353  БАДАМСТЕ55*****  Страхование путешественников  10161883357   
3        21113224  ТТЕ7227715*****  Страхование путешественников    100766461   
4        21113292  ТТЕ7227715*****  Страхование путешественников  10171604758   
...           ...              ...                           ...          ...   
3706  31133782315  ДСЖ-5/1539*****  Страхование путешественников    100649013   
3707  31133836775  ТТЕ7227778*****  Страхование путешественников   1017151314   
3708  31133933241  БТАСЕ275 -*****  Страхование путешественников   1006491844   
3709   3114463583  ТТЕ7227778*****  Страхование путешественников   1006513037   
3710   3114463955  ТТЕ7227715*****  Страхование путешественников  10171539646   

     contract_status     currency_name  duration    country  price  \
0          Действует  Российский рубль        10   Беларусь   1096   
1          Действует  Российский рубль         7  Индонезия   1918   
2          Действует  Российский рубль        10   Беларусь   1096   
3          Действует  Российский рубль        14     Грузия   1534   
4          Действует  Российский рубль         7  Аргентина   1918   
...              ...               ...       ...        ...    ...   
3706       Действует  Российский рубль        15     Турция   2466   
3707       Действует  Российский рубль         7  Индонезия   1918   
3708       Действует  Российский рубль        15     Турция   2466   
3709       Действует  Российский рубль        30     Турция   4932   
3710       Действует  Российский рубль         7  Индонезия   1918   

      insurance_amount  
0              2000000  
1              5000000  
2              2000000  
3              2000000  
4              5000000  
...                ...  
3706           3000000  
3707           5000000  
3708           3000000  
3709           3000000  
3710           5000000  

[3711 rows x 10 columns]

In [4]:
df3 = pd.read_excel('21.10 CASE_CLIENTS.xlsx')
df3.head()

client_id last_name first_name middle_name  age sex
0  10000041307    З*****     Р*****      О*****   30   F
1  10000133158    И*****     М*****      Ю*****   37   M
2    100003773    Е*****     О*****      С*****   50   M
3  10000535718    Е*****     Ф*****      Ч*****   45   M
4  10000583119    И*****     Ф*****      Ф*****   28   M

In [5]:
df4 = pd.read_excel('21.10 experiment_group.xlsx')
df4

client_id    group   experiment_id
0         10001718  control  policy_pricing
1       1017180395  control  policy_pricing
2       1017178047  control  policy_pricing
3       1017174614  control  policy_pricing
4       1017174071  control  policy_pricing
...            ...      ...             ...
14837  11919548695     test  policy_pricing
14838  11919548696     test  policy_pricing
14839  11996903168     test  policy_pricing
14840  11996903169     test  policy_pricing
14841  11996903170     test  policy_pricing

[14842 rows x 3 columns]

In [6]:
result_df = df4.join(other=df2.set_index('client_id'), on='client_id')
result_df

client_id    group   experiment_id   contract_id     contract_num  \
0         10001718  control  policy_pricing  2.123569e+08  ТТЕ7227715*****   
1       1017180395  control  policy_pricing  2.135234e+07  ДСЖ-5/1539*****   
2       1017178047  control  policy_pricing  2.147454e+09  ТМБЕ579 № *****   
3       1017174614  control  policy_pricing  2.135192e+07  ТМГ1535Е7 *****   
4       1017174071  control  policy_pricing  2.162434e+08  ТТЕ7227777*****   
...            ...      ...             ...           ...              ...   
14837  11919548695     test  policy_pricing           NaN              NaN   
14838  11919548696     test  policy_pricing           NaN              NaN   
14839  11996903168     test  policy_pricing           NaN              NaN   
14840  11996903169     test  policy_pricing           NaN              NaN   
14841  11996903170     test  policy_pricing           NaN              NaN   

                       product_name contract_status     currency_name  \
0      Страхование путешественников       Действует  Российский рубль   
1      Страхование путешественников       Действует  Российский рубль   
2      Страхование путешественников       Действует  Российский рубль   
3      Страхование путешественников       Действует  Российский рубль   
4      Страхование путешественников        Завершен  Российский рубль   
...                             ...             ...               ...   
14837                           NaN             NaN               NaN   
14838                           NaN             NaN               NaN   
14839                           NaN             NaN               NaN   
14840                           NaN             NaN               NaN   
14841                           NaN             NaN               NaN   

       duration  country   price  insurance_amount  
0          14.0   Египет  3836.0         5000000.0  
1           7.0  Таиланд  1918.0         5000000.0  
2          14.0   Египет  3836.0         5000000.0  
3           7.0  Таиланд  1918.0         5000000.0  
4          10.0   Канада  1096.0         2000000.0  
...         ...      ...     ...               ...  
14837       NaN      NaN     NaN               NaN  
14838       NaN      NaN     NaN               NaN  
14839       NaN      NaN     NaN               NaN  
14840       NaN      NaN     NaN               NaN  
14841       NaN      NaN     NaN               NaN  

[14842 rows x 12 columns]

In [7]:
result_df1 = result_df.join(other=df.set_index('client_id'), on='client_id')
result_df1

client_id    group   experiment_id   contract_id     contract_num  \
0         10001718  control  policy_pricing  2.123569e+08  ТТЕ7227715*****   
1       1017180395  control  policy_pricing  2.135234e+07  ДСЖ-5/1539*****   
2       1017178047  control  policy_pricing  2.147454e+09  ТМБЕ579 № *****   
3       1017174614  control  policy_pricing  2.135192e+07  ТМГ1535Е7 *****   
4       1017174071  control  policy_pricing  2.162434e+08  ТТЕ7227777*****   
...            ...      ...             ...           ...              ...   
14837  11919548695     test  policy_pricing           NaN              NaN   
14838  11919548696     test  policy_pricing           NaN              NaN   
14839  11996903168     test  policy_pricing           NaN              NaN   
14840  11996903169     test  policy_pricing           NaN              NaN   
14841  11996903170     test  policy_pricing           NaN              NaN   

                       product_name contract_status     currency_name  \
0      Страхование путешественников       Действует  Российский рубль   
1      Страхование путешественников       Действует  Российский рубль   
2      Страхование путешественников       Действует  Российский рубль   
3      Страхование путешественников       Действует  Российский рубль   
4      Страхование путешественников        Завершен  Российский рубль   
...                             ...             ...               ...   
14837                           NaN             NaN               NaN   
14838                           NaN             NaN               NaN   
14839                           NaN             NaN               NaN   
14840                           NaN             NaN               NaN   
14841                           NaN             NaN               NaN   

       duration  country   price  insurance_amount  loss_id loss_name  \
0          14.0   Египет  3836.0         5000000.0      NaN       NaN   
1           7.0  Таиланд  1918.0         5000000.0      NaN       NaN   
2          14.0   Египет  3836.0         5000000.0      NaN       NaN   
3           7.0  Таиланд  1918.0         5000000.0      NaN       NaN   
4          10.0   Канада  1096.0         2000000.0      NaN       NaN   
...         ...      ...     ...               ...      ...       ...   
14837       NaN      NaN     NaN               NaN      NaN       NaN   
14838       NaN      NaN     NaN               NaN      NaN       NaN   
14839       NaN      NaN     NaN               NaN      NaN       NaN   
14840       NaN      NaN     NaN               NaN      NaN       NaN   
14841       NaN      NaN     NaN               NaN      NaN       NaN   

       loss_payout_amt  Unnamed: 4  Unnamed: 5  
0                  NaN         NaN         NaN  
1                  NaN         NaN         NaN  
2                  NaN         NaN         NaN  
3                  NaN         NaN         NaN  
4                  NaN         NaN         NaN  
...                ...         ...         ...  
14837              NaN         NaN         NaN  
14838              NaN         NaN         NaN  
14839              NaN         NaN         NaN  
14840              NaN         NaN         NaN  
14841              NaN         NaN         NaN  

[14842 rows x 17 columns]

In [8]:
del result_df1['Unnamed: 4']
result_df1

client_id    group   experiment_id   contract_id     contract_num  \
0         10001718  control  policy_pricing  2.123569e+08  ТТЕ7227715*****   
1       1017180395  control  policy_pricing  2.135234e+07  ДСЖ-5/1539*****   
2       1017178047  control  policy_pricing  2.147454e+09  ТМБЕ579 № *****   
3       1017174614  control  policy_pricing  2.135192e+07  ТМГ1535Е7 *****   
4       1017174071  control  policy_pricing  2.162434e+08  ТТЕ7227777*****   
...            ...      ...             ...           ...              ...   
14837  11919548695     test  policy_pricing           NaN              NaN   
14838  11919548696     test  policy_pricing           NaN              NaN   
14839  11996903168     test  policy_pricing           NaN              NaN   
14840  11996903169     test  policy_pricing           NaN              NaN   
14841  11996903170     test  policy_pricing           NaN              NaN   

                       product_name contract_status     currency_name  \
0      Страхование путешественников       Действует  Российский рубль   
1      Страхование путешественников       Действует  Российский рубль   
2      Страхование путешественников       Действует  Российский рубль   
3      Страхование путешественников       Действует  Российский рубль   
4      Страхование путешественников        Завершен  Российский рубль   
...                             ...             ...               ...   
14837                           NaN             NaN               NaN   
14838                           NaN             NaN               NaN   
14839                           NaN             NaN               NaN   
14840                           NaN             NaN               NaN   
14841                           NaN             NaN               NaN   

       duration  country   price  insurance_amount  loss_id loss_name  \
0          14.0   Египет  3836.0         5000000.0      NaN       NaN   
1           7.0  Таиланд  1918.0         5000000.0      NaN       NaN   
2          14.0   Египет  3836.0         5000000.0      NaN       NaN   
3           7.0  Таиланд  1918.0         5000000.0      NaN       NaN   
4          10.0   Канада  1096.0         2000000.0      NaN       NaN   
...         ...      ...     ...               ...      ...       ...   
14837       NaN      NaN     NaN               NaN      NaN       NaN   
14838       NaN      NaN     NaN               NaN      NaN       NaN   
14839       NaN      NaN     NaN               NaN      NaN       NaN   
14840       NaN      NaN     NaN               NaN      NaN       NaN   
14841       NaN      NaN     NaN               NaN      NaN       NaN   

       loss_payout_amt  Unnamed: 5  
0                  NaN         NaN  
1                  NaN         NaN  
2                  NaN         NaN  
3                  NaN         NaN  
4                  NaN         NaN  
...                ...         ...  
14837              NaN         NaN  
14838              NaN         NaN  
14839              NaN         NaN  
14840              NaN         NaN  
14841              NaN         NaN  

[14842 rows x 16 columns]

In [9]:
del result_df1['Unnamed: 5']
result_df1

client_id    group   experiment_id   contract_id     contract_num  \
0         10001718  control  policy_pricing  2.123569e+08  ТТЕ7227715*****   
1       1017180395  control  policy_pricing  2.135234e+07  ДСЖ-5/1539*****   
2       1017178047  control  policy_pricing  2.147454e+09  ТМБЕ579 № *****   
3       1017174614  control  policy_pricing  2.135192e+07  ТМГ1535Е7 *****   
4       1017174071  control  policy_pricing  2.162434e+08  ТТЕ7227777*****   
...            ...      ...             ...           ...              ...   
14837  11919548695     test  policy_pricing           NaN              NaN   
14838  11919548696     test  policy_pricing           NaN              NaN   
14839  11996903168     test  policy_pricing           NaN              NaN   
14840  11996903169     test  policy_pricing           NaN              NaN   
14841  11996903170     test  policy_pricing           NaN              NaN   

                       product_name contract_status     currency_name  \
0      Страхование путешественников       Действует  Российский рубль   
1      Страхование путешественников       Действует  Российский рубль   
2      Страхование путешественников       Действует  Российский рубль   
3      Страхование путешественников       Действует  Российский рубль   
4      Страхование путешественников        Завершен  Российский рубль   
...                             ...             ...               ...   
14837                           NaN             NaN               NaN   
14838                           NaN             NaN               NaN   
14839                           NaN             NaN               NaN   
14840                           NaN             NaN               NaN   
14841                           NaN             NaN               NaN   

       duration  country   price  insurance_amount  loss_id loss_name  \
0          14.0   Египет  3836.0         5000000.0      NaN       NaN   
1           7.0  Таиланд  1918.0         5000000.0      NaN       NaN   
2          14.0   Египет  3836.0         5000000.0      NaN       NaN   
3           7.0  Таиланд  1918.0         5000000.0      NaN       NaN   
4          10.0   Канада  1096.0         2000000.0      NaN       NaN   
...         ...      ...     ...               ...      ...       ...   
14837       NaN      NaN     NaN               NaN      NaN       NaN   
14838       NaN      NaN     NaN               NaN      NaN       NaN   
14839       NaN      NaN     NaN               NaN      NaN       NaN   
14840       NaN      NaN     NaN               NaN      NaN       NaN   
14841       NaN      NaN     NaN               NaN      NaN       NaN   

       loss_payout_amt  
0                  NaN  
1                  NaN  
2                  NaN  
3                  NaN  
4                  NaN  
...                ...  
14837              NaN  
14838              NaN  
14839              NaN  
14840              NaN  
14841              NaN  

[14842 rows x 15 columns]

In [10]:
# Курс обмена
exchange_rate = 80

# Применение лямбда-функции для конвертации данных по страховуй сумме
result_df1['insurance_amount'] = result_df1.apply(lambda row: row['insurance_amount'] / exchange_rate if row['currency_name'] == 'Российский рубль' else row['insurance_amount'], axis=1)

result_df1

client_id    group   experiment_id   contract_id     contract_num  \
0         10001718  control  policy_pricing  2.123569e+08  ТТЕ7227715*****   
1       1017180395  control  policy_pricing  2.135234e+07  ДСЖ-5/1539*****   
2       1017178047  control  policy_pricing  2.147454e+09  ТМБЕ579 № *****   
3       1017174614  control  policy_pricing  2.135192e+07  ТМГ1535Е7 *****   
4       1017174071  control  policy_pricing  2.162434e+08  ТТЕ7227777*****   
...            ...      ...             ...           ...              ...   
14837  11919548695     test  policy_pricing           NaN              NaN   
14838  11919548696     test  policy_pricing           NaN              NaN   
14839  11996903168     test  policy_pricing           NaN              NaN   
14840  11996903169     test  policy_pricing           NaN              NaN   
14841  11996903170     test  policy_pricing           NaN              NaN   

                       product_name contract_status     currency_name  \
0      Страхование путешественников       Действует  Российский рубль   
1      Страхование путешественников       Действует  Российский рубль   
2      Страхование путешественников       Действует  Российский рубль   
3      Страхование путешественников       Действует  Российский рубль   
4      Страхование путешественников        Завершен  Российский рубль   
...                             ...             ...               ...   
14837                           NaN             NaN               NaN   
14838                           NaN             NaN               NaN   
14839                           NaN             NaN               NaN   
14840                           NaN             NaN               NaN   
14841                           NaN             NaN               NaN   

       duration  country   price  insurance_amount  loss_id loss_name  \
0          14.0   Египет  3836.0           62500.0      NaN       NaN   
1           7.0  Таиланд  1918.0           62500.0      NaN       NaN   
2          14.0   Египет  3836.0           62500.0      NaN       NaN   
3           7.0  Таиланд  1918.0           62500.0      NaN       NaN   
4          10.0   Канада  1096.0           25000.0      NaN       NaN   
...         ...      ...     ...               ...      ...       ...   
14837       NaN      NaN     NaN               NaN      NaN       NaN   
14838       NaN      NaN     NaN               NaN      NaN       NaN   
14839       NaN      NaN     NaN               NaN      NaN       NaN   
14840       NaN      NaN     NaN               NaN      NaN       NaN   
14841       NaN      NaN     NaN               NaN      NaN       NaN   

       loss_payout_amt  
0                  NaN  
1                  NaN  
2                  NaN  
3                  NaN  
4                  NaN  
...                ...  
14837              NaN  
14838              NaN  
14839              NaN  
14840              NaN  
14841              NaN  

[14842 rows x 15 columns]

In [11]:
# Курс обмена
exchange_rate = 80

# Применение лямбда-функции для конвертации данных по страховуй сумме
result_df1['price'] = result_df1.apply(lambda row: row['price'] / exchange_rate if row['currency_name'] == 'Российский рубль' else row['price'], axis=1)

result_df1

client_id    group   experiment_id   contract_id     contract_num  \
0         10001718  control  policy_pricing  2.123569e+08  ТТЕ7227715*****   
1       1017180395  control  policy_pricing  2.135234e+07  ДСЖ-5/1539*****   
2       1017178047  control  policy_pricing  2.147454e+09  ТМБЕ579 № *****   
3       1017174614  control  policy_pricing  2.135192e+07  ТМГ1535Е7 *****   
4       1017174071  control  policy_pricing  2.162434e+08  ТТЕ7227777*****   
...            ...      ...             ...           ...              ...   
14837  11919548695     test  policy_pricing           NaN              NaN   
14838  11919548696     test  policy_pricing           NaN              NaN   
14839  11996903168     test  policy_pricing           NaN              NaN   
14840  11996903169     test  policy_pricing           NaN              NaN   
14841  11996903170     test  policy_pricing           NaN              NaN   

                       product_name contract_status     currency_name  \
0      Страхование путешественников       Действует  Российский рубль   
1      Страхование путешественников       Действует  Российский рубль   
2      Страхование путешественников       Действует  Российский рубль   
3      Страхование путешественников       Действует  Российский рубль   
4      Страхование путешественников        Завершен  Российский рубль   
...                             ...             ...               ...   
14837                           NaN             NaN               NaN   
14838                           NaN             NaN               NaN   
14839                           NaN             NaN               NaN   
14840                           NaN             NaN               NaN   
14841                           NaN             NaN               NaN   

       duration  country   price  insurance_amount  loss_id loss_name  \
0          14.0   Египет  47.950           62500.0      NaN       NaN   
1           7.0  Таиланд  23.975           62500.0      NaN       NaN   
2          14.0   Египет  47.950           62500.0      NaN       NaN   
3           7.0  Таиланд  23.975           62500.0      NaN       NaN   
4          10.0   Канада  13.700           25000.0      NaN       NaN   
...         ...      ...     ...               ...      ...       ...   
14837       NaN      NaN     NaN               NaN      NaN       NaN   
14838       NaN      NaN     NaN               NaN      NaN       NaN   
14839       NaN      NaN     NaN               NaN      NaN       NaN   
14840       NaN      NaN     NaN               NaN      NaN       NaN   
14841       NaN      NaN     NaN               NaN      NaN       NaN   

       loss_payout_amt  
0                  NaN  
1                  NaN  
2                  NaN  
3                  NaN  
4                  NaN  
...                ...  
14837              NaN  
14838              NaN  
14839              NaN  
14840              NaN  
14841              NaN  

[14842 rows x 15 columns]

In [12]:
# Курс обмена
exchange_rate = 80

# Применение лямбда-функции для конвертации данных по страховуй сумме
result_df1['loss_payout_amt'] = result_df1.apply(lambda row: row['loss_payout_amt'] / exchange_rate if row['currency_name'] == 'Российский рубль' else row['loss_payout_amt'], axis=1)

result_df1

client_id    group   experiment_id   contract_id     contract_num  \
0         10001718  control  policy_pricing  2.123569e+08  ТТЕ7227715*****   
1       1017180395  control  policy_pricing  2.135234e+07  ДСЖ-5/1539*****   
2       1017178047  control  policy_pricing  2.147454e+09  ТМБЕ579 № *****   
3       1017174614  control  policy_pricing  2.135192e+07  ТМГ1535Е7 *****   
4       1017174071  control  policy_pricing  2.162434e+08  ТТЕ7227777*****   
...            ...      ...             ...           ...              ...   
14837  11919548695     test  policy_pricing           NaN              NaN   
14838  11919548696     test  policy_pricing           NaN              NaN   
14839  11996903168     test  policy_pricing           NaN              NaN   
14840  11996903169     test  policy_pricing           NaN              NaN   
14841  11996903170     test  policy_pricing           NaN              NaN   

                       product_name contract_status     currency_name  \
0      Страхование путешественников       Действует  Российский рубль   
1      Страхование путешественников       Действует  Российский рубль   
2      Страхование путешественников       Действует  Российский рубль   
3      Страхование путешественников       Действует  Российский рубль   
4      Страхование путешественников        Завершен  Российский рубль   
...                             ...             ...               ...   
14837                           NaN             NaN               NaN   
14838                           NaN             NaN               NaN   
14839                           NaN             NaN               NaN   
14840                           NaN             NaN               NaN   
14841                           NaN             NaN               NaN   

       duration  country   price  insurance_amount  loss_id loss_name  \
0          14.0   Египет  47.950           62500.0      NaN       NaN   
1           7.0  Таиланд  23.975           62500.0      NaN       NaN   
2          14.0   Египет  47.950           62500.0      NaN       NaN   
3           7.0  Таиланд  23.975           62500.0      NaN       NaN   
4          10.0   Канада  13.700           25000.0      NaN       NaN   
...         ...      ...     ...               ...      ...       ...   
14837       NaN      NaN     NaN               NaN      NaN       NaN   
14838       NaN      NaN     NaN               NaN      NaN       NaN   
14839       NaN      NaN     NaN               NaN      NaN       NaN   
14840       NaN      NaN     NaN               NaN      NaN       NaN   
14841       NaN      NaN     NaN               NaN      NaN       NaN   

       loss_payout_amt  
0                  NaN  
1                  NaN  
2                  NaN  
3                  NaN  
4                  NaN  
...                ...  
14837              NaN  
14838              NaN  
14839              NaN  
14840              NaN  
14841              NaN  

[14842 rows x 15 columns]

In [13]:
result_df1['group'].unique()

array(['control', 'test', nan], dtype=object)

In [15]:
df6 = result_df1[result_df1['group'] == 'control']
df6

client_id    group   experiment_id   contract_id     contract_num  \
0         10001718  control  policy_pricing  2.123569e+08  ТТЕ7227715*****   
1       1017180395  control  policy_pricing  2.135234e+07  ДСЖ-5/1539*****   
2       1017178047  control  policy_pricing  2.147454e+09  ТМБЕ579 № *****   
3       1017174614  control  policy_pricing  2.135192e+07  ТМГ1535Е7 *****   
4       1017174071  control  policy_pricing  2.162434e+08  ТТЕ7227777*****   
...            ...      ...             ...           ...              ...   
11716  10113431702  control  policy_pricing           NaN              NaN   
11721  10113471682  control  policy_pricing           NaN              NaN   
11723  10113481998  control  policy_pricing           NaN              NaN   
11725  10113496169  control  policy_pricing           NaN              NaN   
11727  10113496171  control  policy_pricing           NaN              NaN   

                       product_name contract_status     currency_name  \
0      Страхование путешественников       Действует  Российский рубль   
1      Страхование путешественников       Действует  Российский рубль   
2      Страхование путешественников       Действует  Российский рубль   
3      Страхование путешественников       Действует  Российский рубль   
4      Страхование путешественников        Завершен  Российский рубль   
...                             ...             ...               ...   
11716                           NaN             NaN               NaN   
11721                           NaN             NaN               NaN   
11723                           NaN             NaN               NaN   
11725                           NaN             NaN               NaN   
11727                           NaN             NaN               NaN   

       duration  country   price  insurance_amount  loss_id loss_name  \
0          14.0   Египет  47.950           62500.0      NaN       NaN   
1           7.0  Таиланд  23.975           62500.0      NaN       NaN   
2          14.0   Египет  47.950           62500.0      NaN       NaN   
3           7.0  Таиланд  23.975           62500.0      NaN       NaN   
4          10.0   Канада  13.700           25000.0      NaN       NaN   
...         ...      ...     ...               ...      ...       ...   
11716       NaN      NaN     NaN               NaN      NaN       NaN   
11721       NaN      NaN     NaN               NaN      NaN       NaN   
11723       NaN      NaN     NaN               NaN      NaN       NaN   
11725       NaN      NaN     NaN               NaN      NaN       NaN   
11727       NaN      NaN     NaN               NaN      NaN       NaN   

       loss_payout_amt  
0                  NaN  
1                  NaN  
2                  NaN  
3                  NaN  
4                  NaN  
...                ...  
11716              NaN  
11721              NaN  
11723              NaN  
11725              NaN  
11727              NaN  

[6043 rows x 15 columns]

In [16]:
df6_1 = df6[df6['price'] > 0]
df6_1 # датафрейм для контрольной группы, которые заключили договор страхования

client_id    group   experiment_id   contract_id     contract_num  \
0         10001718  control  policy_pricing  2.123569e+08  ТТЕ7227715*****   
1       1017180395  control  policy_pricing  2.135234e+07  ДСЖ-5/1539*****   
2       1017178047  control  policy_pricing  2.147454e+09  ТМБЕ579 № *****   
3       1017174614  control  policy_pricing  2.135192e+07  ТМГ1535Е7 *****   
4       1017174071  control  policy_pricing  2.162434e+08  ТТЕ7227777*****   
...            ...      ...             ...           ...              ...   
11262  10101515185  control  policy_pricing  2.121139e+07  БАЕАМСТЕ35*****   
11371  10105147478  control  policy_pricing  2.121719e+07  ТТЕ7227779*****   
11543  10109591977  control  policy_pricing  2.136212e+08  ИТЕ7 67777*****   
11670  10111664837  control  policy_pricing  2.136416e+08  БТАСЕ273 -*****   
11709  10113069118  control  policy_pricing  2.136798e+07  БТАСЕ276 -*****   

                           product_name contract_status     currency_name  \
0          Страхование путешественников       Действует  Российский рубль   
1          Страхование путешественников       Действует  Российский рубль   
2          Страхование путешественников       Действует  Российский рубль   
3          Страхование путешественников       Действует  Российский рубль   
4          Страхование путешественников        Завершен  Российский рубль   
...                                 ...             ...               ...   
11262      Страхование путешественников       Действует  Российский рубль   
11371  Страхование путешественников USD       Действует        Доллар США   
11543      Страхование путешественников       Действует  Российский рубль   
11670      Страхование путешественников       Действует  Российский рубль   
11709      Страхование путешественников       Действует  Российский рубль   

       duration  country     price  insurance_amount  loss_id  \
0          14.0   Египет   47.9500           62500.0      NaN   
1           7.0  Таиланд   23.9750           62500.0      NaN   
2          14.0   Египет   47.9500           62500.0      NaN   
3           7.0  Таиланд   23.9750           62500.0      NaN   
4          10.0   Канада   13.7000           25000.0      NaN   
...         ...      ...       ...               ...      ...   
11262     365.0   Турция  750.0000           37500.0      NaN   
11371      21.0      США  115.0000          100000.0      NaN   
11543      23.0   Египет   78.7625           62500.0     21.0   
11670      24.0   Турция   82.1875           62500.0      NaN   
11709      25.0   Турция   85.6125           62500.0      NaN   

                         loss_name  loss_payout_amt  
0                              NaN              NaN  
1                              NaN              NaN  
2                              NaN              NaN  
3                              NaN              NaN  
4                              NaN              NaN  
...                            ...              ...  
11262                          NaN              NaN  
11371                          NaN              NaN  
11543  Оказание медицинской помощи           1250.0  
11670                          NaN              NaN  
11709                          NaN              NaN  

[1949 rows x 15 columns]

In [17]:
df7 = result_df1[result_df1['group'] == 'test']
df7

client_id group   experiment_id   contract_id     contract_num  \
1854   10111468947  test  policy_pricing  2.164933e+09  ТТЕ7227715*****   
1855   10191188103  test  policy_pricing  2.153493e+07  ДСЖ-5/1539*****   
1856   10111378413  test  policy_pricing  2.164890e+09  ДСЖ-5/1539*****   
1857   10105565615  test  policy_pricing  2.121749e+08  ДСЖ-5/1539*****   
1858   10107579603  test  policy_pricing  2.135595e+08  ТТЕ7227779*****   
...            ...   ...             ...           ...              ...   
14837  11919548695  test  policy_pricing           NaN              NaN   
14838  11919548696  test  policy_pricing           NaN              NaN   
14839  11996903168  test  policy_pricing           NaN              NaN   
14840  11996903169  test  policy_pricing           NaN              NaN   
14841  11996903170  test  policy_pricing           NaN              NaN   

                       product_name contract_status     currency_name  \
1854   Страхование путешественников       Действует  Российский рубль   
1855   Страхование путешественников       Действует  Российский рубль   
1856   Страхование путешественников       Действует  Российский рубль   
1857   Страхование путешественников       Действует  Российский рубль   
1858   Страхование путешественников       Действует  Российский рубль   
...                             ...             ...               ...   
14837                           NaN             NaN               NaN   
14838                           NaN             NaN               NaN   
14839                           NaN             NaN               NaN   
14840                           NaN             NaN               NaN   
14841                           NaN             NaN               NaN   

       duration  country    price  insurance_amount  loss_id loss_name  \
1854       19.0   Турция  65.0625           62500.0      NaN       NaN   
1855       14.0   Египет  47.9500           62500.0      NaN       NaN   
1856       18.0   Турция  61.6500           62500.0      NaN       NaN   
1857       10.0  Армения  13.7000           25000.0      NaN       NaN   
1858       22.0   Турция  75.3375           62500.0      NaN       NaN   
...         ...      ...      ...               ...      ...       ...   
14837       NaN      NaN      NaN               NaN      NaN       NaN   
14838       NaN      NaN      NaN               NaN      NaN       NaN   
14839       NaN      NaN      NaN               NaN      NaN       NaN   
14840       NaN      NaN      NaN               NaN      NaN       NaN   
14841       NaN      NaN      NaN               NaN      NaN       NaN   

       loss_payout_amt  
1854               NaN  
1855               NaN  
1856               NaN  
1857               NaN  
1858               NaN  
...                ...  
14837              NaN  
14838              NaN  
14839              NaN  
14840              NaN  
14841              NaN  

[6306 rows x 15 columns]

In [18]:
df7_1 = df7[df7['price'] > 0]
df7_1  # датафрейм для тестовой группы, которые заключили договор страхования

client_id group   experiment_id   contract_id     contract_num  \
1854   10111468947  test  policy_pricing  2.164933e+09  ТТЕ7227715*****   
1855   10191188103  test  policy_pricing  2.153493e+07  ДСЖ-5/1539*****   
1856   10111378413  test  policy_pricing  2.164890e+09  ДСЖ-5/1539*****   
1857   10105565615  test  policy_pricing  2.121749e+08  ДСЖ-5/1539*****   
1858   10107579603  test  policy_pricing  2.135595e+08  ТТЕ7227779*****   
...            ...   ...             ...           ...              ...   
11638  10111515943  test  policy_pricing  2.136361e+08  ТТЕ7227715*****   
11642  10111547681  test  policy_pricing  2.136392e+08  ДСЖ-5/1539*****   
11674  10111754681  test  policy_pricing  2.136638e+08  ТТЕ7227778*****   
11705  10111991766  test  policy_pricing  2.136755e+08  ДСЖ-5/1539*****   
14724  10193891479  test  policy_pricing  2.153797e+07  ЕТЕ7215377*****   

                           product_name contract_status     currency_name  \
1854       Страхование путешественников       Действует  Российский рубль   
1855       Страхование путешественников       Действует  Российский рубль   
1856       Страхование путешественников       Действует  Российский рубль   
1857       Страхование путешественников       Действует  Российский рубль   
1858       Страхование путешественников       Действует  Российский рубль   
...                                 ...             ...               ...   
11638      Страхование путешественников       Действует  Российский рубль   
11642      Страхование путешественников       Действует  Российский рубль   
11674      Страхование путешественников       Действует  Российский рубль   
11705      Страхование путешественников       Действует  Российский рубль   
14724  Страхование путешественников USD       Действует        Доллар США   

       duration  country    price  insurance_amount  loss_id  \
1854       19.0   Турция  65.0625           62500.0      NaN   
1855       14.0   Египет  47.9500           62500.0      NaN   
1856       18.0   Турция  61.6500           62500.0      NaN   
1857       10.0  Армения  13.7000           25000.0      NaN   
1858       22.0   Турция  75.3375           62500.0      NaN   
...         ...      ...      ...               ...      ...   
11638      24.0   Турция  82.1875           62500.0      NaN   
11642      24.0   Турция  82.1875           62500.0      NaN   
11674      24.0   Турция  82.1875           62500.0      NaN   
11705      25.0      ОАЭ  85.6125           62500.0     24.0   
14724      15.0      США  82.0000          100000.0      NaN   

                         loss_name  loss_payout_amt  
1854                           NaN              NaN  
1855                           NaN              NaN  
1856                           NaN              NaN  
1857                           NaN              NaN  
1858                           NaN              NaN  
...                            ...              ...  
11638                          NaN              NaN  
11642                          NaN              NaN  
11674                          NaN              NaN  
11705  Оказание медицинской помощи          18750.0  
14724                          NaN              NaN  

[1667 rows x 15 columns]

In [17]:
# Проводим T-тест
stat, p = ttest_ind(df6_1['price'], df7_1['price'])

print('Статистика T =', stat)
print('p-значение =', p)

if p < 0.05:
    print('Отвергаем нулевую гипотезу: средние значения выборок статистически значимо различаются.')
else:
    print('Не отвергаем нулевую гипотезу: нет статистически значимых различий между средними значениями выборок.')

Статистика T = -2.5991455015086387
p-значение = 0.009383612311978436
Отвергаем нулевую гипотезу: средние значения выборок статистически значимо различаются.


In [18]:
# Проводим тест Шапиро-Уилка
stat, p_value = shapiro(df6_1['price'])

print('Статистика теста:', stat)
print('p-значение:', p_value)

alpha = 0.05
if p_value > alpha:
    print('Данные можно считать нормально распределенными')
else:
    print('Данные не являются нормально распределенными')

Статистика теста: 0.1974509358406067
p-значение: 0.0
Данные не являются нормально распределенными


In [19]:
# Проводим тест Шапиро-Уилка
stat, p_value = shapiro(df7_1['price'])

print('Статистика теста:', stat)
print('p-значение:', p_value)

alpha = 0.05
if p_value > alpha:
    print('Данные можно считать нормально распределенными')
else:
    print('Данные не являются нормально распределенными')

Статистика теста: 0.2944965958595276
p-значение: 0.0
Данные не являются нормально распределенными


In [20]:
# Проводим тест Манна — Уитни для не нормально распределенных выборок при альтернативной гипотезе , что выботки отличаются
stat5, p5 = stats.mannwhitneyu(df6_1['price'], df7_1['price'])

print('Статистика T =', stat5)
print('p-значение =', p5)

if p5 < 0.05:
    print('Отвергаем нулевую гипотезу: средние значения выборок статистически значимо различаются.')
else:
    print('Не отвергаем нулевую гипотезу: нет статистически значимых различий между средними значениями выборок.')

Статистика T = 1843303.0
p-значение = 1.397411195174327e-12
Отвергаем нулевую гипотезу: средние значения выборок статистически значимо различаются.


In [21]:
df6_1['price'].describe() # статистика контрольной группы

count    1949.000000
mean       51.931946
std       141.189247
min        13.700000
25%        19.175000
50%        24.662500
75%        47.950000
max      2000.000000
Name: price, dtype: float64

In [22]:
df7_1['price'].describe() # статистика тестовой группы

count    1667.000000
mean       65.348425
std       169.188975
min         6.850000
25%        13.700000
50%        19.175000
75%        61.650000
max      2000.000000
Name: price, dtype: float64

In [35]:
df7_1['loss_payout_amt'].sum() - df7_1['price'].sum() # убыток тестовой группы

-16560.82500000001

In [36]:
df6_1['loss_payout_amt'].sum() - df6_1['price'].sum() # убыток контрольной группы

-21465.362500000003

In [27]:
len(df6_1) / (len(result_df1) - len(df7)) # конверсия контрольной группы

0.22832708528584816

In [28]:
len(df7_1) / len(df7) # конверсия тестовой группы

0.26435141135426576

In [31]:
df6_1['price'].mode() 

0    47.95
dtype: float64

In [32]:
df7_1['price'].mode() 

0    13.7
dtype: float64

In [33]:
df7_1['price'].count() - df6_1['price'].count() 

-282

In [34]:
df7_1['price'].sum() - df6_1['price'].sum() 

7720.462500000009

# для контрольной группы средняя цена полиса 51.93$ , а  наиболее частая цена 47.95 $ . для тестовой группы средняя цена полиса 65.34$ , а наиболее частая цена 13.7 $ . Что говорит у тестовой группы цена полиса более дешевая для большинства клиентов, а у выброчных клиетов цена высокая. Не смотря , что количество клиентов в тестовой группе на 282 меньше, сумма за полисы на 7720$ больше.